iroh-net/src/magic_endpoint.rs

```diff
@@ -422,6 +438,21 @@ impl MagicEndpoint {
         Ok(())
     }
 
+    /// Get addressing information of this [`MagicEndpoint`].
+    pub async fn my_addr_info(&self) -> Result<AddrInfo> {
```

Comment:
I see that `magicsock/endpoint.rs` now has `addr_info` which returns exactly what we return here. Maybe route to that to have 1, not 2 async calls to the actor here. But can also be a followup.